pkg/runtime/handler.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+
+	pbd "istio.io/api/mixer/v1/config/descriptor"
+	"istio.io/mixer/pkg/adapter"
+	pb "istio.io/mixer/pkg/config/proto"
+	"istio.io/mixer/pkg/expr"
+	"istio.io/mixer/pkg/template"
+)
+
+type (
+	// BuilderInfoFinder is used to find specific handlers BuilderInfo for configuration.
+	BuilderInfoFinder func(name string) (*adapter.BuilderInfo, bool)
+
+	// HandlerFactory builds adapter.Handler object from adapter and instances configuration.
+	HandlerFactory interface {
+		Build(*pb.Handler, []*pb.Instance, adapter.Env) (adapter.Handler, error)
+	}
+
+	handlerFactory struct {
+		tmplRepo          template.Repository
+		typeChecker       expr.TypeChecker
+		attrDescFinder    expr.AttributeDescriptorFinder
+		builderInfoFinder BuilderInfoFinder
+
+		// protects cache
+		lock           sync.RWMutex
```

Comment:
This is executed on the config change path. BuildHandler () only gets called when a particular handler needs to be rebuilt. I supposed this could be useful when Mixer is loading...
Ok we can leave it in, but no code is going to use at it present.